shared/src/commonMain/kotlin/io/customer/shared/tracking/model/Task.kt

```diff
@@ -0,0 +1,12 @@
+package io.customer.shared.tracking.model
+
+/**
+ * Data class to hold tracking event attributes to track.
+ *
+ * @property profileIdentifier the id used for identifying user profile/
+ * @property activity tracking activity details.
+ */
+internal data class Task(
```

Comment:
Task is not an open class now, the tracking event are moved into newer class Activity (copied naming convention from Activity Logs). Activity is a sealed interface which means only events added to it can be tracked. But since we are planning to avoid a lot of tables and put its json in DB, having limited activity classes is convenient to manage. Network layer, however, will map to its own classes before passing events to the api.

Reply:
This means we are not storing values separately from `Activity` class in Database. Instead, keeping its json dump for now. But there are plenty of other attributes e.g. timestamp, identifiers, etc. that will be stored in database using SQL. And tasks filtering, status updates will be done using SQL queries. Also, 

> Why we opted for merging json with DB

We don't really want to complicate this in the first phase. All activities may have some unique attributes and keeping all of them separately means a lot of tables. And with these numbers of tables, we'll not only need to take care of a lot of foreign keys, joins, migrations, cascading options, but also, would still have to dump json somewhere as our custom attributes are open (supports Any). So we see more complications than benefit for this. You can see sample table proposal in [bq/draft](https://github.com/customerio/mobile-shared/blob/bq/draft/shared/src/commonMain/sqldelight/local/TrackingTask.sq).